packages/client/src/pages/GamePage.tsx

```diff
@@ -1,15 +1,47 @@
-import React from 'react'
+import { useState, createContext } from 'react';
 import GameLayout from '../layouts/GameLayout'
 import ActionLayer from '../layers/ActionLayer'
 import LandscapeLayer from '../layers/LandscapeLayer'
+import GamePause from '../components/gamePause'
+import GameOver from '../components/GameOver'
+import Engine from '../engine/Engine';
+
+const GameContext = createContext({});
 
 const GamePage = () => {
-  return (
-    <GameLayout>
-      <LandscapeLayer />
-      <ActionLayer />
-    </GameLayout>
+	const [pauseVisible, setPauseVisible] = useState(false);
+	const [gameOverVisible, setGameOverVisible] = useState(false);
+
+	function handlePause(){
+		setPauseVisible(true)
+	}
+	
+	function handleContinue(){
+		setPauseVisible(false)
+		const engine = Engine.get()
+		engine.pause(false)
+	}
+
+	function handleGameOver(){
+		setGameOverVisible(true)
+	}
+	
+	function handleNewGame(){
+		setGameOverVisible(false)
+		const engine = Engine.get()
+		engine.start()
+	}
+
+	return (
+		<GameLayout>
+			<GamePause visible={pauseVisible} handleClose={handleContinue} outSideClickEnable/>
+			<GameOver visible={gameOverVisible} handleClose={handleNewGame} />
+			<LandscapeLayer />
+			<ActionLayer {...{handlePause, handleGameOver}}/>
```

Comment:
В чём выгода sread оператора тут, и почему не проще это передать как в остальных местах?
```suggestion
			<ActionLayer handlePause={handlePause} handleGameOver={handleGameOver} />
```

Reply:
О... Я не сразу к этому пришёл и знатно навозился с типизацией каждого колбэка на каждом этапе цепочки.
Передаю объект с набором колбэков а не по отдельности чтобы проще было их прокидывать дальше.
Можно было, конечно обернуть в контекст, но учитывая что целью является уже следующий компонент то посчитал излишним.